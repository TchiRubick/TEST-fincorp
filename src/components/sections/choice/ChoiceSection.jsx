import React from 'react';

import './ChoiceSection.scss';

import Container from '../../container/Container';
import CardOptions from '../../options/card/CardOptions';

import home from '../../../assets/img/home-insurance-1.png';
import car from '../../../assets/img/car-insurance-1.png';
import life from '../../../assets/img/life-insurance-1.png';
import business from '../../../assets/img/business-insurance-1.png';
import travel from '../../../assets/img/travel-insurance-1.png';
import other from '../../../assets/img/other-insurance-1.png';

const choiceList = [
  {
    image: home,
    title: 'Home Insurance',
    content: 'Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.',
    color: '#aff3ff',
  },
  {
    image: car,
    title: 'Car Insurance',
    content: 'Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.',
    color: '#fbd1ff',
  },
  {
    image: life,
    title: 'Life Insurance',
    content: 'Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.',
    color: '#daffd3',
  },
  {
    image: business,
    title: 'Business Insurance',
    content: 'Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.',
    color: '#fff5b3',
  },
  {
    image: travel,
    title: 'Travel Insurance',
    content: 'Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.',
    color: '#ffdad2',
  },
  {
    image: other,
    title: 'Other Insurance',
    content: 'Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.',
    color: '#d2d3ff',
  },
];

const ChoiceSection = () => {
  const renderListCard = () => choiceList.map((c) => (
    <CardOptions title={c.title} image={c.image} color={c.color}>
      {c.content}
    </CardOptions>
  ));

  return (
    <Container>
      <div className="section-choice">
        <div className="section-choice__title">
          <span className="section-choice__main">
            Choose your Insurance
          </span>
          <span className="section-choice__sub">
            Keep Your Life Smile, Safe, and Wealthy
          </span>
        </div>
        <div className="section-choice__content">
          {renderListCard()}
        </div>
      </div>
    </Container>
  );
};

export default ChoiceSection;
