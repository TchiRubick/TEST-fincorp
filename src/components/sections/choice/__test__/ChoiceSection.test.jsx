import React from 'react';
import ReactDom from 'react-dom';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import ChoiceSection from '../ChoiceSection';

afterEach(cleanup);

it('renders good', () => {
  const div = document.createElement('div');

  ReactDom.render(<ChoiceSection />, div);
});

it('as expected snapshot 1', () => {
  const tree = renderer.create(<ChoiceSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
