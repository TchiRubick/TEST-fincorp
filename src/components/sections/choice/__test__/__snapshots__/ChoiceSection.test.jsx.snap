// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as expected snapshot 1 1`] = `
<div
  className="comp-container"
>
  <div
    className="section-choice"
  >
    <div
      className="section-choice__title"
    >
      <span
        className="section-choice__main"
      >
        Choose your Insurance
      </span>
      <span
        className="section-choice__sub"
      >
        Keep Your Life Smile, Safe, and Wealthy
      </span>
    </div>
    <div
      className="section-choice__content"
    >
      <div
        className="comp-card-options"
      >
        <div
          className="comp-card-options__image"
        >
          <img
            alt="Home Insurance_image"
            src="home-insurance-1.png"
            style={
              Object {
                "backgroundColor": "#aff3ff",
                "height": "6rem",
                "width": "6rem",
              }
            }
          />
        </div>
        <div
          className="comp-card-options__title"
        >
          <span
            className="comp-card-options__title-text"
          >
            Home Insurance
          </span>
        </div>
        <div
          className="comp-card-options__text"
        >
          Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.
        </div>
      </div>
      <div
        className="comp-card-options"
      >
        <div
          className="comp-card-options__image"
        >
          <img
            alt="Car Insurance_image"
            src="car-insurance-1.png"
            style={
              Object {
                "backgroundColor": "#fbd1ff",
                "height": "6rem",
                "width": "6rem",
              }
            }
          />
        </div>
        <div
          className="comp-card-options__title"
        >
          <span
            className="comp-card-options__title-text"
          >
            Car Insurance
          </span>
        </div>
        <div
          className="comp-card-options__text"
        >
          Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.
        </div>
      </div>
      <div
        className="comp-card-options"
      >
        <div
          className="comp-card-options__image"
        >
          <img
            alt="Life Insurance_image"
            src="life-insurance-1.png"
            style={
              Object {
                "backgroundColor": "#daffd3",
                "height": "6rem",
                "width": "6rem",
              }
            }
          />
        </div>
        <div
          className="comp-card-options__title"
        >
          <span
            className="comp-card-options__title-text"
          >
            Life Insurance
          </span>
        </div>
        <div
          className="comp-card-options__text"
        >
          Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.
        </div>
      </div>
      <div
        className="comp-card-options"
      >
        <div
          className="comp-card-options__image"
        >
          <img
            alt="Business Insurance_image"
            src="business-insurance-1.png"
            style={
              Object {
                "backgroundColor": "#fff5b3",
                "height": "6rem",
                "width": "6rem",
              }
            }
          />
        </div>
        <div
          className="comp-card-options__title"
        >
          <span
            className="comp-card-options__title-text"
          >
            Business Insurance
          </span>
        </div>
        <div
          className="comp-card-options__text"
        >
          Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.
        </div>
      </div>
      <div
        className="comp-card-options"
      >
        <div
          className="comp-card-options__image"
        >
          <img
            alt="Travel Insurance_image"
            src="travel-insurance-1.png"
            style={
              Object {
                "backgroundColor": "#ffdad2",
                "height": "6rem",
                "width": "6rem",
              }
            }
          />
        </div>
        <div
          className="comp-card-options__title"
        >
          <span
            className="comp-card-options__title-text"
          >
            Travel Insurance
          </span>
        </div>
        <div
          className="comp-card-options__text"
        >
          Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.
        </div>
      </div>
      <div
        className="comp-card-options"
      >
        <div
          className="comp-card-options__image"
        >
          <img
            alt="Other Insurance_image"
            src="other-insurance-1.png"
            style={
              Object {
                "backgroundColor": "#d2d3ff",
                "height": "6rem",
                "width": "6rem",
              }
            }
          />
        </div>
        <div
          className="comp-card-options__title"
        >
          <span
            className="comp-card-options__title-text"
          >
            Other Insurance
          </span>
        </div>
        <div
          className="comp-card-options__text"
        >
          Insurance can have various effects on society through the way that it changes who bears the cost of losses and damage.
        </div>
      </div>
    </div>
  </div>
</div>
`;
